### ISSUE TITLE
Bad Blotchy DDS Artifacts

### ISSUE THREAD

**Issue by ReeceBar** [photopea/photopea#3183]
I've black and whited an image, and notice that exporting it it has gained blotchy green artifacts. The copy is the new one, notice how it looks far worse than the original, all I did was black and whited it.
 I think this has something to do with some bad compression Photopea is doing with DDS files?
[Flashlight DDS.zip](https://github.com/photopea/photopea/files/6374297/Flashlight.DDS.zip)

**Comment by photopea** [comment]
I see green artifacts already in your file "flashlight.dds".

![image](https://user-images.githubusercontent.com/15955558/116112555-f4219680-a6b7-11eb-93b5-55b50e01cb62.png)

DDS is a lossy format, just like JPG. You lose quality with each export to DDS or JPG. And while JPG supports 100% quality mode, DDS does not (there is always a loss).

Why don't you use PNG as the main format, and convert to DDS only once?

**Comment by ReeceBar** [comment]
Well I've tried that, using an image upscaler that smoothed it, but when I reconverted it back to dds it looked far worse than the original 512 dds file for some reason Ill try to redo that and send pics

**Comment by ReeceBar** [comment]
[New folder.zip](https://github.com/photopea/photopea/files/6380218/New.folder.zip)
 You can see that the new dds that has been made to a png black and whited and upscaled by an ai upscaler to 1024x looks worse than the original 512x dds when converted to one for some reason. The one on the left is the original, the greenish blotchier one is the right one.

**Comment by photopea** [comment]
why do you upscale it?

As I said, you always lose quality when converting to DDS.

Could you provide a PNG of what you want to achieve, a DDS of that PNG from Photopea, and another DDS of the same PNG from some other program, to prove, that a better-looking DDS is possible to achieve?

**Comment by photopea** [comment]
So I looked into it and I found an error, which made the output quality worse than it could be.

The output quality of DDS should be better now. Is it ok now?

**Comment by photopea** [comment]
My new method guarantees, that if you open a DDS in Photopea and save it back as a DDS (without any editing), the resulting file will look completely identical to the original DDS.

**Comment by ReeceBar** [comment]
I think it might be less greenish, but it's still more pixelated than the original
![Screenshot (403)](https://user-images.githubusercontent.com/42558495/116236123-ac892200-a72c-11eb-9ac7-1b8d289fe228.png)

**Comment by ReeceBar** [comment]
Oh, you accidently closed it. Should I open up a new issue?

**Comment by photopea** [comment]
I think it is not possible to convert your PNG into a better looking DDS. DDS always removes some quality.

If you disagree, could you create a DDS, which looks better than a DDS from Photopea?

**Comment by ReeceBar** [comment]
So even when I converted a DDS to PNG, upscaled it, smoothed it, then convert it back to DDS, despite being double the resolution it will always look far worse? Theirs nothing that can be done in photopea to make it a better looking higher res dds files?

**Comment by photopea** [comment]
As I said, create a "better loking" DDS without Photopea, so that we have a proof, that a better looking DDS file can exist. Do not use Photopea at all.

Once I see, that some other program works better than Photopea, I will reseach it and improve Photopea.

**Comment by ReeceBar** [comment]
![Screenshot (409)](https://user-images.githubusercontent.com/42558495/116480664-bf484780-a84f-11eb-94ea-1298a1e0c106.png)
 There a better looking DDS from Gimp done the same way I did when trying to get a better looking DDS from Photopea. Gimps is a lot less greenish and blotchy, it's actually smooth, not pixelated. 
[Gimp1024x.zip](https://github.com/photopea/photopea/files/6395164/Gimp1024x.zip)

**Comment by ReeceBar** [comment]
The one on the left is from photopea, just screenshot the screenshot of it lol. The one on the right is Gimp

**Comment by photopea** [comment]
I opened your DDS from Gimp in Photopea and it contains transparency. Is that correct? How exactly should the DDS from Gimp be displayed?

**Comment by ReeceBar** [comment]
Actually Gimp seems to have broken the DDS file, I'm looking into if I can prevent Gimp from inverting the flashlight, and removing the background as thats what Gimp is doing when exporting DDS files for some reason here's Gimp's broken DDS converted to png

**Comment by ReeceBar** [comment]
![Gimp1024x](https://user-images.githubusercontent.com/42558495/116776036-887f5680-aa34-11eb-9b6b-290d692154d5.png)

**Comment by photopea** [comment]
There was a bug in Photopea when opening your DDS from Gimp. It should look well now.

So does the DDS from GIMP work in your game? Is it what you want Photopea to give you?

Gimp gives an uncompressed grayscale image (it is bigger than the DDS from Photopea, and it would be 3x more big, if it contained color).

**Comment by ReeceBar** [comment]
I don't understand why but the DDS files Gimp produces are red ingame, instead of white. Everything looks fine, even loads fine in Photopea and Gimp, as in it looks completely grey, no odd colors and it's a lot smoother, almost no pixilation, it just doesn't load right ingame.
 Would like to see a no compression option for DDS files in Photopea, that might allow me to achieve what I'm trying to do. Anyway, here's the file gimp produced that looks fine but is red ingame 
[flashlightgobo01.zip](https://github.com/photopea/photopea/files/6410475/flashlightgobo01.zip)

**Comment by photopea** [comment]
You can disable the compression when exporting a DDS in Photopea now.

If your game does not accept a DDS file from Gimp, I think it is a bug in that game, you should report it to the creators of a game.
